File: Maths/Palindrome.cpp
#include<iostream>
#include<stdbool.h>
#include<cstring>
using namespace std;
//For String
bool Palindrome(string str){
    string rev;
    int len=str.length();
    for(int i=len-1;i>=0;i--){
        rev += str[i];
    }
    if (rev == str) {
        cout << "The string is a palindrome" << endl;
        return true;
    } else {
        cout << "Not a palindrome" << endl;
        return false;
    }
}

//For Int
bool isPalindrome(int x){
    int rev=0;
    int temp=x;
    if(x<0)
        return false;
    while(x){
        rev=rev*10+x%10;
        x/=10;
    }
    if(temp==rev){
    cout<<"The no is palindrome"<<endl;
    }
    else{
        cout<<"Not palindrome"<<endl;
    }
}
int main(){
    int x;
    string x1;
    cout<<"Enter no to check:"<<endl;
    cin>>x;
    isPalindrome(x);
    cout<<"Enter string to check:"<<endl;
    cin>>x1;
    Palindrome(x1);
    return 0;
}

File: Maths -- Geometry/C#/Palindrome.cs
public class Solution {
    public bool IsPalindrome(int x) {
        if(x<0){
            return false;
        }
        int num = x;
        int check = 0;
        while(num!=0){
            int rem = num%10;
            check = check*10 + rem;
            num/=10;
        }
        if(check==x){
            return true;
        }
        else{
            return false;
        }
    }
}